hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/TokenKeyTranslator.java

```diff
@@ -76,6 +81,7 @@ private Key getTokenKey(Token token, int keyType) throws InvalidTransactionExcep
             case 16 -> token.supplyKey();
             case 32 -> token.feeScheduleKey();
             case 64 -> token.pauseKey();
+            case 128 -> metadataSupport ? token.metadataKey() : null;
```

Comment:
is returning null the right thing here.  if the config is off should we not signal some sort of error?

Reply:
As we want to keep the same behaviour of ignoring this case when the flag is disabled, returning null complies with the `default` value of the switch and the token check before it.